content/posts/147-react-context/index.md

```diff
@@ -288,6 +288,42 @@ function UserInfo() {
 }
 ```
 
+#### 3.2.1 Updating the context value from inside the consumer
+
+The React Context API does not provide a dedicated method to update the context value from consumer components, but this can be easily implemented by providing an update function right in the context next to the value itself.
+
+```jsx
+import { createContext, useState, useContext } from 'react';
+
+const UserContext = createContext({
+  userName: '',
+  setUserName: () => {},
+});
+
+function Application() {
+  const [userName, setUserName] = useState('John Smith');
+  const value = { userName, setUserName };
+  return (
+    <UserContext.Provider value={value}>
```

Comment:
Every time Application re-renders the context value (which is created from an object literal) is going to change triggering all consumers to re-render. Please memoize `value` by `userName`. See more details here: https://reactjs.org/docs/context.html#caveats.

Also write an explanation of why the memoization was added.